cfsdk/src/main/java/io/harness/cfsdk/cloud/analytics/AnalyticsManager.java

```diff
@@ -92,14 +92,19 @@ public AnalyticsManager(
         this.networkChecker = networkChecker;
 
         final long frequencyMs = config.getMetricsPublishingIntervalInMillis();
-        scheduledExecutorService.scheduleAtFixedRate(this::postMetricsThread,frequencyMs/2, frequencyMs, TimeUnit.MILLISECONDS);
+        scheduledExecutorService.scheduleWithFixedDelay(this::postMetricsThread,frequencyMs/2, frequencyMs, TimeUnit.MILLISECONDS);
```

Comment:
The previous method was deprecated and could cause issues with caching of threads.